thirdeye/thirdeye-frontend/app/pods/components/rootcause-metric-select/component.js

```diff
@@ -20,6 +28,26 @@ export default Ember.Component.extend({
 
   placeholder: 'Search for a Metric',
 
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    const { selectedUrn, selectedUrnCache } = this.getProperties('selectedUrn', 'selectedUrnCache');
+
+    if (!_.isEqual(selectedUrn, selectedUrnCache)) {
+      this.set('selectedUrnCache', selectedUrn);
+
+      if (selectedUrn) {
+        const url = `/data/metric/${selectedUrn.split(':')[2]}`;
+        fetch(url)
+          .then(res => res.json())
+          .then(res => this.set('selectedMetric', res));
+
+      } else {
+        this.set('selectedMetric', undefined);
```

Comment:
Is there any way we could refactor this piece of code so that there's only one this.setProperties at the bottom. 
Having multiple consecutive `set` usually results in a slight decrease in performance (it's minor, but it adds up)

Reply:
the else clause here is defensive programming, i.e. it should never actually be called during execution.